ArduCopter/events.cpp

```diff
@@ -7,29 +7,33 @@
         return;
     }
 
     if (should_disarm_on_failsafe()) {
         arming.disarm();
     } else {
-        if (control_mode == Mode::Number::AUTO && g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION) {
-            // continue mission
-        } else if (flightmode->is_landing() &&
-                   battery.has_failsafed() &&
-                   battery.get_highest_failsafe_priority() <= FAILSAFE_LAND_PRIORITY) {
-            // continue landing or other high priority failsafes
+        if (flightmode->is_landing() && ((battery.has_failsafed() && battery.get_highest_failsafe_priority() <= FAILSAFE_LAND_PRIORITY) || (g2.fs_options & ParametersG2::FS_OPT_THR_LAND_CONTINUE))) {
+            // Allow landing to continue because battery failsafe requires it or FS_OPTIONS is set to continue landing
+            gcs().send_text(MAV_SEVERITY_WARNING, "RADIO FAILSAFE - CONTINUING LANDING");
+        } else if (control_mode == Mode::Number::AUTO && (g2.fs_options & ParametersG2::FS_OPT_THR_MISSION_CONTINUE)) {
+            // Allow mission to continue because FS_OPTIONS is set to continue mission
+            gcs().send_text(MAV_SEVERITY_WARNING, "RADIO FAILSAFE - CONTINUING MISSION");       
+        } else if (control_mode == Mode::Number::GUIDED && (g2.fs_options & ParametersG2::FS_OPT_THR_GUIDED_CONTINUE) && (g.failsafe_gcs != FS_GCS_DISABLED)) {
+            // Allow guided mode to continue because FS_THR_OPTION is set to continue in guided mode.  Only if the GCS failsafe is enabled.
+            gcs().send_text(MAV_SEVERITY_WARNING, "RADIO FAILSAFE - CONTINUING GUIDED MODE");
+        } else if (RC_Channels::has_active_overrides() && (g2.fs_options & ParametersG2::FS_OPT_THR_CONTINUE_RCOVERRIDE) && (g.failsafe_gcs != FS_GCS_DISABLED)) {
+            // Allow RC Overrides to continue because FS_THR_OPTION is set to continue in RC Overirides. Only if the GCS failsafe is enabled.
+            gcs().send_text(MAV_SEVERITY_WARNING, "RADIO FAILSAFE - CONTINUING RC OVERRIDE");
         } else {
-            if (g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_RTL) {
-                set_mode_RTL_or_land_with_pause(MODE_REASON_RADIO_FAILSAFE);
-            } else if (g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION) {
+            if (g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_RTL || g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION) {
                 set_mode_RTL_or_land_with_pause(MODE_REASON_RADIO_FAILSAFE);
             } else if (g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_SMARTRTL_OR_RTL) {
                 set_mode_SmartRTL_or_RTL(MODE_REASON_RADIO_FAILSAFE);
             } else if (g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_SMARTRTL_OR_LAND) {
                 set_mode_SmartRTL_or_land_with_pause(MODE_REASON_RADIO_FAILSAFE);
-            } else { // g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_LAND
+            } else if (g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_LAND) {
```

Comment:
this is a behaviour change

Reply:
Fixed to once again handle invalid values going to land.